pynumdiff/total_variation_regularization/_total_variation_regularization.py

```diff
@@ -1,123 +1,112 @@
-    :param gamma: (float) regularization parameter
-    :param solver: (string) Solver to use. Solver options include: 'MOSEK' and 'CVXOPT',
-                            in testing, 'MOSEK' was the most robust.
-    :return: x_hat    : estimated (smoothed) x
-             dxdt_hat : estimated derivative of x
+def _total_variation_regularized_derivative(x, dt, order, gamma, solver=solver):
+    """Generalized total variation regularized derivatives. Use convex optimization (cvxpy) to solve for a
+    total variation regularized derivative. Default solver is MOSEK: , which is not
+    always available. Fall back to CVXPY's default.
+
+    :param np.array[float] x: data to differentiate
+    :param float dt: time step
+    :param int order: 1, 2, or 3, the derivative to regularize
+    :param float gamma: regularization parameter
+    :param str solver: Solver to use. Solver options include: 'MOSEK', 'CVXOPT', 'CLARABEL', 'ECOS'.
+                            In testing, 'MOSEK' was the most robust.
+  
+    :return: tuple[np.array, np.array] of\n
+             - **x_hat** -- estimated (smoothed) x
+             - **dxdt_hat** -- estimated derivative of x
     """
-
     # Normalize
     mean = np.mean(x)
     std = np.std(x)
     x = (x-mean)/std
 
     # Define the variables for the highest order derivative and the integration constants
-    var = cvxpy.Variable(len(x) + N)
+    deriv_values = cvxpy.Variable(len(x)) # values of the order^th derivative, in which we're penalizing variation
+    integration_constants = cvxpy.Variable(order) # constants of integration that help get us back to x
 
     # Recursively integrate the highest order derivative to get back to the position
-    derivatives = [var[N:]]
-    for i in range(N):
-        d = cvxpy.cumsum(derivatives[-1]) + var[i]
-        derivatives.append(d)
-
-    # Compare the recursively integration position to the noisy position
-    sum_squared_error = cvxpy.sum_squares(derivatives[-1] - x)
-
-    # Total variation regularization on the highest order derivative
-    r = cvxpy.sum(gamma*cvxpy.tv(derivatives[0]))
+    y = deriv_values
+    for i in range(order):
+        y = cvxpy.cumsum(y) + integration_constants[i]
 
     # Set up and solve the optimization problem
-    obj = cvxpy.Minimize(sum_squared_error + r)
-    prob = cvxpy.Problem(obj)
+    prob = cvxpy.Problem(cvxpy.Minimize(
+        # Compare the recursively integrated position to the noisy position, and add TVR penalty
+        cvxpy.sum_squares(y - x) + cvxpy.sum(gamma*cvxpy.tv(deriv_values)) ))
     prob.solve(solver=solver)
 
-    # Recursively calculate the value of each derivative
-    final_derivative = var.value[N:]
-    derivative_values = [final_derivative]
-    for i in range(N):
-        d = np.cumsum(derivative_values[-1]) + var.value[i]
-        derivative_values.append(d)
-    for i, _ in enumerate(derivative_values):
-        derivative_values[i] = derivative_values[i]/(dt**(N-i))
```

Comment:
You were going through and properly normalizing all the derivatives' values, but we only use the 1st derivative, so now I loop through just freely forgetting values until the 1st derivative, and then I only divide by `dt` for that one.